### PR TITLE
Added manager 2.0 installation

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
~~- [ ] I added the relevant `backport` labels~~
~~- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)~~
~~- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
~~- [ ] I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: jenkins-pipelines/manager-sanity.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+longevityPipeline(
+    params: params,
+
+    backend: 'aws',
+    aws_region: '''["us-east-1", "us-west-2"]''',
+    test_name: 'mgmt_cli_test.MgmtCliTest.test_manager_sanity',
+    test_config: 'test-cases/manager/manager-regression-multiDC-set-distro.yaml',
+
+    timeout: [time: 500, unit: 'MINUTES'],
+    post_behaviour: 'destroy'
```
Comment (fruch) [inline jenkins-pipelines/manager-sanity.jenkinsfile]: remove this one, and replace with the correct default you want for this job

File: jenkins-pipelines/manager-sanity.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!groovy
+
+// trick from https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/43
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+longevityPipeline(
+    params: params,
+
+    backend: 'aws',
```
Comment (fruch) [inline jenkins-pipelines/manager-sanity.jenkinsfile]: add the correct ip_ssh_connections default for multi-region

File: sdcm/cluster.py [hunk 3]
```diff
@@ -1903,13 +1917,24 @@ def install_mgmt(self, scylla_repo, scylla_mgmt_repo):
...
+
         if self.is_docker():
             self.remoter.run('sudo supervisorctl restart scylla-manager')
             res = self.remoter.run('sudo supervisorctl status scylla-manager')
         else:
             self.remoter.run('sudo systemctl restart scylla-manager.service')
             res = self.remoter.run('sudo systemctl status scylla-manager.service')
 
+        if self.is_rhel_like():
+            configuring_new_manager_command = dedent("""
```
Comment (fruch) [inline sdcm/cluster.py]: what's `new` has todo with this parameter 
also add a TODO comment regarding debian/ubuntu like installations

File: sdcm/cluster.py [hunk 4]
```diff
@@ -1903,13 +1917,24 @@ def install_mgmt(self, scylla_repo, scylla_mgmt_repo):
         """.format(mgmt_user=mgmt_user, rsa_id_dst=rsa_id_dst, rsa_id_dst_pub=rsa_id_dst_pub))  # generate ssh public key from private key.
         self.remoter.run('sudo bash -cxe "%s"' % ssh_config_script)
 
+        self.config_scylla_manager_yaml(segments_per_repair=100)
```
Comment (fruch) [inline sdcm/cluster.py]: isn't this something we want to control ? i.e. in  SCT configuration
also if we hardcode this value, please comment on why and what it's intended for.

**Comment by fruch** [comment]
One of the two options for this PR:
1) change the default to a stable manager version from manager branch-2.0
2) change the code to support manager 1.4

without one of those two, we'll be failing on manager on any longevity that would run out of master.

File: sdcm/cluster.py [hunk 5]
```diff
@@ -3503,12 +3532,24 @@ def node_setup(self, node, **kwargs):  # pylint: disable=unused-argument
...
+    def install_scylla_manager(self, node, auth_token):
+        self.mgmt_auth_token = auth_token  # pylint: disable=attribute-defined-outside-init
         if self.params.get('use_mgmt', default=None):
             node.install_mgmt(scylla_repo=self.params.get('scylla_repo_m'),
-                              scylla_mgmt_repo=self.params.get('scylla_mgmt_repo'))
+                              scylla_mgmt_repo=self.params.get('scylla_mgmt_repo'), auth_token=auth_token)
+            wait.wait_for(func=self.wait_until_manager_is_up, step=20, text='Waiting until the manager client is up',
+                          timeout=300, throw_exc=True)
+
+    def wait_until_manager_is_up(self):
```
Comment (fruch) [inline sdcm/cluster.py]: this is not the function that does the waiting, it's just doing the check
`is_manager_up()`

File: sdcm/mgmt.py [hunk 6]
```diff
@@ -626,19 +626,15 @@ def add_cluster(self, name, host=None, db_cluster=None, client_encrypt=None, use
...
-                                                                                            ssh_user, name)
+        cmd = 'cluster add --host={}  --name={} --auth-token {}'.format(
+            host, name, auth_token)
         # Adding client-encryption parameters if required
         if client_encrypt:
             if not db_cluster:
                 LOGGER.warning("db_cluster is not given. Scylla-Manager connection to cluster may "
                                "fail since not using client-encryption parameters.")
-            else:  # check if scylla-node has client-encrypt
+            else:  # check if scylla-node has client-encrypts
```
Comment (fruch) [inline sdcm/mgmt.py]: what's `encrypts` ?

File: test-cases/manager/manager-regression-multiDC-set-distro.yaml [hunk 7]
```diff
@@ -4,7 +4,7 @@ instance_type_db: 'i3.large'
 instance_type_loader: 'c4.large'
 instance_type_monitor: 't3.small'
 
-region_name: 'eu-west-1 us-east-1'
+region_name: 'us-east-1 us-west-2'
```
Comment (fruch) [inline test-cases/manager/manager-regression-multiDC-set-distro.yaml]: why move to us only regions ?

File: vars/longevityPipeline.groovy [hunk 8]
```diff
@@ -109,6 +110,11 @@ def call(Map pipelineParams) {
                                     export SCT_TAG_AMI_WITH_RESULT="${params.tag_ami_with_result}"
                                     export SCT_IP_SSH_CONNECTIONS="${params.ip_ssh_connections}"
 
+                                    if [[ ! -z "${params.scylla_mgmt_repo}" ]] ; then
+                                    export SCT_SCYLLA_MGMT_REPO="${params.scylla_mgmt_repo}"
```
Comment (fruch) [inline vars/longevityPipeline.groovy]: can you indent this line ? for better readability ?

File: test-cases/manager/manager-regression-multiDC-set-distro.yaml (hunk 7 above)
Comment (ShlomiBalalis) [reply test-cases/manager/manager-regression-multiDC-set-distro.yaml]: To lower the load on Ireland

**Comment by fruch** [comment]
waiting for final confirmation from @ShlomiBalalis that this is working as expected, before merging this

File: sdcm/mgmt.py [hunk 9]
```diff
@@ -749,14 +747,23 @@ def __init__(self, manager_node):
...
+            #     # setting SCTOOL_ASCII_TABLES so output of sctool will be in ascii
+            #     res = self._remoter_run(cmd="sudo bash -cxe 'SCTOOL_ASCII_TABLES=TRUE\nsctool {}'".format(cmd))
+            # else:
             res = self._remoter_run(cmd='sudo sctool {}'.format(cmd))
             LOGGER.debug("sctool output: %s", res.stdout)
         except (UnexpectedExit, Failure) as ex:
             raise ScyllaManagerError("Encountered an error on sctool command: {}: {}".format(cmd, ex))
 
+        if replace_broken_unicode_values:
+            res.stdout = res.stdout.replace('���', '│')
```
Comment (fruch) [inline sdcm/mgmt.py]: this is the solution for this issue ?

**Comment by fruch** [comment]
@ShlomiBalalis, you seem to keep changing this PR, ping me when you think you are done with this one.

File: sdcm/cluster.py [hunk 10]
```diff
@@ -1864,13 +1878,24 @@ def install_mgmt(self, scylla_repo, scylla_mgmt_repo):
...
         else:
             self.remoter.run('sudo systemctl restart scylla-manager.service')
             res = self.remoter.run('sudo systemctl status scylla-manager.service')
 
+        if self.is_rhel_like():  # TODO: Add debian and ubuntu support
+            configuring_manager_command = dedent("""
+            scyllamgr_ssl_cert_gen
+            sed -i 's/#tls_cert_file/tls_cert_file/' /etc/scylla-manager/scylla-manager.yaml
+            sed -i 's/#tls_key_file/tls_key_file/' /etc/scylla-manager/scylla-manager.yaml
+            systemctl restart scylla-manager
```
Comment (roydahan) [inline sdcm/cluster.py]: Only this command may be different on other distros like deb and ubuntu.
Move the if here and handle the ubuntu and debian as well.
Don't leave the TODO here.

File: sdcm/cluster.py [hunk 11]
```diff
@@ -3457,12 +3486,25 @@ def node_setup(self, node, **kwargs):  # pylint: disable=unused-argument
...
+            self.install_scylla_manager(node, auth_token=kwargs["auth_token"])
 
-    def install_scylla_manager(self, node):
+    def install_scylla_manager(self, node, auth_token):
+        self.mgmt_auth_token = auth_token  # pylint: disable=attribute-defined-outside-init
         if self.params.get('use_mgmt', default=None):
             node.install_mgmt(scylla_repo=self.params.get('scylla_repo_m'),
-                              scylla_mgmt_repo=self.params.get('scylla_mgmt_repo'))
+                              scylla_mgmt_repo=self.params.get('scylla_mgmt_repo'), auth_token=auth_token,
+                              segments_per_repair=self.params.get('mgmt_segments_per_repair', 10))
```
Comment (roydahan) [inline sdcm/cluster.py]: why is this "10" here?

File: sdcm/mgmt.py (hunk 9 above)
Comment (roydahan) [reply sdcm/mgmt.py]: For now, because I don't want to be blocked by that.

File: sdcm/cluster.py (hunk 10 above)
Comment (ShlomiBalalis) [reply sdcm/cluster.py]: @roydahan It's more than this one command, I didn't test the manager on debian yet and I don't think I should merge it without testing

Comment (fruch) [reply sdcm/cluster.py]: @ShlomiBalalis please create a trello task for doing the manager debian/ubuntu support

File: mgmt_cli_test.py [hunk 12]
```diff
@@ -58,24 +58,16 @@ def test_mgmt_cluster_crud(self):
         hosts = self._get_cluster_hosts_ip()
         selected_host = hosts[0]
         cluster_name = 'mgr_cluster_crud'
-        mgr_cluster = manager_tool.add_cluster(name=cluster_name, host=selected_host)
```
Comment (yarongilor) [inline mgmt_cli_test.py]: didn't we agree to support multiple manager versions, using the new decorator?

Comment (yarongilor) [reply mgmt_cli_test.py]: @bentsi ? @fruch ? do i miss something here?

**Pull request merged**